pkg/nsx/services/vpc/vpc.go

```diff
@@ -185,13 +184,29 @@ func InitializeVPC(service common.Service) (*VPCService, error) {
 	return VPCService, nil
 }
 
+func (s *VPCService) GetVPCsByFreshNamespace(ctx context.Context, namespace string) []*model.Vpc {
+	namespaceObj, sharedNamespace, err := s.resolveSharedVPCNamespace(ctx, namespace)
+	if err != nil {
+		log.Error(err, "Failed to get Namespace")
+		return nil
+	}
+	if sharedNamespace == nil {
+		return s.VpcStore.GetVPCsByNamespaceIDFromStore(string(namespaceObj.UID))
+	}
+	return s.VpcStore.GetVPCsByNamespaceIDFromStore(string(sharedNamespace.UID))
+}
+
 func (s *VPCService) GetVPCsByNamespace(ctx context.Context, namespace string) []*model.Vpc {
-	sns, err := s.getSharedVPCNamespaceFromNS(ctx, namespace)
+	// only used in NetworkInfo controller, no need to check whether is a shared Namespace
```

Comment:
what this comment meaning? inside this func, we still need to check if is shared Namespace obj?